src/main/java/com/kodilla/ecommercee/service/CartService.java

```diff
@@ -0,0 +1,73 @@
+package com.kodilla.ecommercee.service;
+
+import com.kodilla.ecommercee.domain.Cart;
+import com.kodilla.ecommercee.domain.ProductItem;
+import com.kodilla.ecommercee.domain.dto.CartDto;
+import com.kodilla.ecommercee.domain.dto.OrderDto;
+import com.kodilla.ecommercee.mapper.CartMapper;
+import com.kodilla.ecommercee.repository.CartRepository;
+import com.kodilla.ecommercee.repository.OrderRepository;
+import com.kodilla.ecommercee.repository.ProductItemRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class CartService {
+
+    @Autowired
+    private CartRepository cartRepository;
+
+    @Autowired
+    private OrderRepository orderRepository;
+
+    @Autowired
+    private ProductItemRepository productItemRepository;
+
+    @Autowired
+    private CartMapper cartMapper;
+
+    public Cart createCart(@RequestParam long userId) {
+        return new Cart();
+    }
+
+    public Optional<Cart> getCart(final long id) { //ProductItem zamiast Cart
+        return cartRepository.findById(id);
+    }
+
+    public Optional<Cart> addItemToCard(@RequestParam long cartId, @RequestBody ProductItem productItem) {
```

Comment:
Hmm Czy aby napewno trzeba przekazywać Ciało obiektu typu ProductItem ? Do koszyka będą dodawane rzeczy które juz istnieją w sklepie, a nie całkowicie nowe obiekty. Więc zamiast przekazywania nowego obiektu typu ProductItem przekazałbym Id jakiegoś productItemu a następnie w metodzie znaleźć go przy pomocy serwisu i dodać do koszyka. Co o tym sądzisz ?

Reply:
To co jest w koszyku w obiekcie ProductItem zawiera Listę produktów (List<Product>) które pochodzą z bazy danych, 
oraz dodtakowe informacje, których nie ma w klasie Product, czyli BigDecimal ammount i int quantity. Nie ma sensu 
rozszerzać klasy Product o te dodatkowe pola w prawdziwym sklepie koszyk zawiera listę produktów oraz dodatkowe informacje.